server/src/permissions/index.ts

```diff
@@ -156,24 +89,29 @@ export const permissions = shield({
   Mutation: {
     signup: rules.isAdmin,
 
+    createAccessControl: rules.isAdmin,
+
     createSource: rules.isAuthenticatedUser,
-    deleteSource: rules.isWriter,
+    deleteSource: rules.isSourceWriter,
+
+    upsertCredential: rules.isSourceWriter,
+    deleteCredential: rules.isSourceWriter,
 
-    createResource: rules.isWriter,
-    updateResource: rules.isWriter,
-    deleteResource: rules.isWriter,
+    createResource: rules.isSourceWriter,
+    updateResource: rules.isSourceWriter,
+    deleteResource: rules.isSourceWriter,
 
-    createAttribute: rules.isWriter,
-    updateAttribute: rules.isWriter,
+    createAttribute: rules.isSourceWriter,
+    updateAttribute: rules.isSourceWriter,
     updateComments: rules.isAuthenticatedUser,
```

Comment:
shouldn't this be `isSourceReader` ?

Reply:
In my mind, everybody authenticated could update comments. Do you think I should add a new rule?